### PR TITLE
[rotel] New channel to select FM/DAB/iRadio preset

### PR DESCRIPTION
**Pull request by lolodomo**
For models RCX-1500, RDG-1520, RT-11, RT-1570, T11 and T14

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java [hunk 1]
```diff
@@ -1488,9 +1549,23 @@ public void onNewMessageEvent(EventObject event) {
...
                     updateChannelState(CHANNEL_SOURCE);
                     updateChannelState(CHANNEL_MAIN_SOURCE);
+                    RotelCommand presetGetCmd = getRadioPresetGetCommand(source);
+                    if (presetGetCmd != null) {
+                        // Request current preset (with a delay)
+                        scheduler.schedule(() -> {
+                            try {
+                                sendCommand(presetGetCmd);
+                            } catch (RotelException e) {
+                            }
```
Comment (jlaur) [inline bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1563]: Consider adding a comment here why this exception is ignored.

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java [hunk 2]
```diff
@@ -1327,6 +1367,27 @@ public void onNewMessageEvent(EventObject event) {
...
+            try {
+                preset = Integer.parseInt(key.substring(KEY_DAB_PRESET.length()));
+            } catch (NumberFormatException e) {
+            }
+            key = KEY_DAB_PRESET;
+        } else if (key.startsWith(KEY_IRADIO_PRESET)) {
+            try {
+                preset = Integer.parseInt(key.substring(KEY_IRADIO_PRESET.length()));
+            } catch (NumberFormatException e) {
+            }
```
Comment (jlaur) [inline bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1387]: Consider adding a comment here why this exception is ignored.

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java [hunk 3]
```diff
@@ -1327,6 +1367,27 @@ public void onNewMessageEvent(EventObject event) {
...
+            try {
+                preset = Integer.parseInt(key.substring(KEY_FM_PRESET.length()));
+            } catch (NumberFormatException e) {
+            }
+            key = KEY_FM_PRESET;
+        } else if (key.startsWith(KEY_DAB_PRESET)) {
+            try {
+                preset = Integer.parseInt(key.substring(KEY_DAB_PRESET.length()));
+            } catch (NumberFormatException e) {
+            }
```
Comment (jlaur) [inline bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1381]: Consider adding a comment here why this exception is ignored.

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java [hunk 4]
```diff
@@ -1327,6 +1367,27 @@ public void onNewMessageEvent(EventObject event) {
             default:
                 break;
         }
+        int preset = 0;
+        if (key.startsWith(KEY_FM_PRESET)) {
+            try {
+                preset = Integer.parseInt(key.substring(KEY_FM_PRESET.length()));
+            } catch (NumberFormatException e) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1374]: Consider adding a comment here why this exception is ignored.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only minor comments towards swallowed exceptions.

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1387]: Done

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java (hunk 3 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1381]: Done

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java (hunk 4 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1374]: Done

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java:1563]: If I remove the try/catch, I have an error about unhandled exception.
So I added a log.

**Comment by lolodomo** [comment]
@jlaur : this is ready.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
Thank you

**Pull request merged**